pool.ts

```diff
@@ -0,0 +1,109 @@
+import { dial } from "deno";
+import { Client } from "./client.ts";
+import { Connection } from "./connection.ts";
+import { ConnectionParams, IConnectionParams } from "./connection_params.ts";
+import { Query, QueryConfig, QueryResult } from "./query.ts";
+import { defer, Deferred } from "./deps.ts";
+
+export class Pool {
+  private _connectionParams: IConnectionParams;
+  private _connections: Array<Connection>;
+  private _availableConnections: DeferredStack<Connection>;
+  private _size: number;
+  private _ready: Promise<void>;
+
+  constructor(connectionParams: IConnectionParams, size: number) {
+    this._connectionParams = connectionParams;
+    this._size = size;
+    this._ready = this.startup();
+  }
+
+  private async newConnection(): Promise<Connection> {
+    const { host, port } = this._connectionParams;
+    let addr = `${host}:${port}`;
+
+    const conn = await dial("tcp", addr);
+    const connection = new Connection(conn, this._connectionParams);
+
+    await connection.startup({ ...this._connectionParams });
+    await connection.initSQL();
+    return connection;
+  }
+
+  get size(): number {
+    return this._size;
+  }
+  get available(): number {
+    return this._availableConnections.size;
+  }
+
+  private async startup(): Promise<void> {
+    const connecting = [...Array(this.size)].map(
```

Comment:
can you change this method to create new instances of `Client` instead of connection? That way API of `Pool` and `Client` will be pretty much consistent (only difference is that `Pool` knows how to handle busy/idle clients).

 I have pretty neat idea in mind, that we'll extend `Client` class to `PoolClient`, but that's for another PR.

Reply:
Added `PooledClient`, which I think solves the constructor issue too.